### PR TITLE
Add get immv def func

### PR DESCRIPTION
**Pull request by thoshiai**
This function can reconstruct the underlying SELECT command for a IMMV. (This is a decompiled reconstruction, not the original text of the command)

### CODE REVIEW

File: createas.c [hunk 1]
```diff
@@ -1203,7 +1203,7 @@ CreateIndexOnIMMV(Query *query, Relation matviewRel, bool is_create)
 					(errmsg("could not create an index on immv \"%s\" automatically",
 							RelationGetRelationName(matviewRel)),
 					 errdetail("This target list does not have all the primary key columns, "
-							   "or this view does not contain GROUP BY or DISTINCT clause."),
+							   "or this view does not contain DISTINCT clause."),
```
Comment (yugo-n) [inline createas.c:1206]: Is this change necessary?

File: pg_ivm.c [hunk 2]
```diff
@@ -320,3 +322,30 @@ PgIvmImmvPrimaryKeyIndexId(void)
...
+get_immv_def(PG_FUNCTION_ARGS)
+{
+	Oid	matviewOid = PG_GETARG_OID(0);
+	Relation matviewRel = NULL;
+	Query *query = NULL;
+	char *querystring = NULL;
+
+	matviewRel = table_open(matviewOid, AccessShareLock);
+	/* Make sure IMMV is a table. */
+	Assert(matviewRel->rd_rel->relkind == RELKIND_RELATION);
```
Comment (yugo-n) [inline pg_ivm.c:338]: Due to this assert, passing a relation other than table causes an error.
```
postgres=# \d v
                  View "public.v"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 i      | integer |           |          | 

postgres=# select get_immv_def('v');
TRAP: FailedAssertion("matviewRel->rd_rel->relkind == RELKIND_RELATION", File: "pg_ivm.c", Line: 338, PID: 37560)
```

How about just return NULL in this case, as pg_get_viewdef does?

File: pg_ivm.c [hunk 3]
```diff
@@ -320,3 +322,30 @@ PgIvmImmvPrimaryKeyIndexId(void)
...
+
+	query = get_immv_query(matviewRel);
+	if (query == NULL)
+	{
+		table_close(matviewRel, NoLock);
+		PG_RETURN_TEXT_P(cstring_to_text(""));
```
Comment (yugo-n) [inline pg_ivm.c:344]: Should we return NULL instead of an empty string, as pg_get_viewdef?

File: pg_ivm.c [hunk 4]
```diff
@@ -320,3 +322,30 @@ PgIvmImmvPrimaryKeyIndexId(void)
...
+	}
+
+	querystring = pg_ivm_get_querydef(get_immv_query(matviewRel), false);
```
Comment (yugo-n) [inline pg_ivm.c:347]: Why we need call get_immv_query again?

**Comment by thoshiai** [comment]
Thank you for comments. I fix it and rebased latest version

File: createas.c (hunk 1 above)
Comment (thoshiai) [reply createas.c]: Sorry, this fix is not related by this commit. so I revert it

File: pg_ivm.c (hunk 2 above)
Comment (thoshiai) [reply pg_ivm.c]: OK I change to NULL

File: pg_ivm.c (hunk 3 above)
Comment (thoshiai) [reply pg_ivm.c]: ok, I fixed it

**Comment by yugo-n** [comment]
Thanks! It seams good.
